// @aws-sdk/client-route-53/dist-es/commands/DeleteKeySigningKeyCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getIdNormalizerPlugin } from "@aws-sdk/middleware-sdk-route53";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteKeySigningKeyRequestFilterSensitiveLog, DeleteKeySigningKeyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlDeleteKeySigningKeyCommand, serializeAws_restXmlDeleteKeySigningKeyCommand, } from "../protocols/Aws_restXml";
export class DeleteKeySigningKeyCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, DeleteKeySigningKeyCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getIdNormalizerPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "Route53Client";
        const commandName = "DeleteKeySigningKeyCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DeleteKeySigningKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteKeySigningKeyResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlDeleteKeySigningKeyCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlDeleteKeySigningKeyCommand(output, context);
    }
}
