// @aws-sdk/client-route-53/dist-es/commands/GetHealthCheckCountCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetHealthCheckCountRequestFilterSensitiveLog, GetHealthCheckCountResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlGetHealthCheckCountCommand, serializeAws_restXmlGetHealthCheckCountCommand, } from "../protocols/Aws_restXml";
export class GetHealthCheckCountCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, GetHealthCheckCountCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "Route53Client";
        const commandName = "GetHealthCheckCountCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetHealthCheckCountRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetHealthCheckCountResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlGetHealthCheckCountCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlGetHealthCheckCountCommand(output, context);
    }
}
