// @aws-sdk/client-route-53/dist-es/commands/ListHostedZonesByVPCCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListHostedZonesByVPCRequestFilterSensitiveLog, ListHostedZonesByVPCResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlListHostedZonesByVPCCommand, serializeAws_restXmlListHostedZonesByVPCCommand, } from "../protocols/Aws_restXml";
export class ListHostedZonesByVPCCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListHostedZonesByVPCCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "Route53Client";
        const commandName = "ListHostedZonesByVPCCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListHostedZonesByVPCRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListHostedZonesByVPCResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlListHostedZonesByVPCCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlListHostedZonesByVPCCommand(output, context);
    }
}
