// @aws-sdk/middleware-sdk-route53/dist-cjs/change-resource-record-sets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getChangeResourceRecordSetsPlugin = exports.changeResourceRecordSetsMiddlewareOptions = exports.changeResourceRecordSetsMiddleware = void 0;
const constants_1 = require("./constants");
function changeResourceRecordSetsMiddleware() {
    return (next) => async (args) => {
        const { ChangeBatch } = args.input;
        const Changes = [];
        for (const change of ChangeBatch.Changes) {
            const { AliasTarget } = change.ResourceRecordSet;
            if (AliasTarget) {
                Changes.push({
                    ...change,
                    ResourceRecordSet: {
                        ...change.ResourceRecordSet,
                        AliasTarget: {
                            ...AliasTarget,
                            HostedZoneId: AliasTarget.HostedZoneId.replace(constants_1.IDENTIFIER_PREFIX_PATTERN, ""),
                        },
                    },
                });
            }
            else {
                Changes.push(change);
            }
        }
        return next({
            ...args,
            input: {
                ...args.input,
                ChangeBatch: {
                    ...ChangeBatch,
                    Changes,
                },
            },
        });
    };
}
exports.changeResourceRecordSetsMiddleware = changeResourceRecordSetsMiddleware;
exports.changeResourceRecordSetsMiddlewareOptions = {
    step: "initialize",
    tags: ["ROUTE53_IDS", "CHANGE_RESOURCE_RECORD_SETS"],
    name: "changeResourceRecordSetsMiddleware",
    override: true,
};
const getChangeResourceRecordSetsPlugin = (unused) => ({
    applyToStack: (clientStack) => {
        clientStack.add(changeResourceRecordSetsMiddleware(), exports.changeResourceRecordSetsMiddlewareOptions);
    },
});
exports.getChangeResourceRecordSetsPlugin = getChangeResourceRecordSetsPlugin;
