// @aws-sdk/middleware-sdk-route53/dist-cjs/id-normalizer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getIdNormalizerPlugin = exports.idNormalizerMiddlewareOptions = exports.idNormalizerMiddleware = void 0;
const constants_1 = require("./constants");
const IDENTIFIER_PARAMETERS = ["DelegationSetId", "HostedZoneId", "Id"];
function idNormalizerMiddleware() {
    return (next) => async (args) => {
        const input = { ...args.input };
        for (const paramName of IDENTIFIER_PARAMETERS) {
            const param = input[paramName];
            if (param) {
                input[paramName] = param.replace(constants_1.IDENTIFIER_PREFIX_PATTERN, "");
            }
        }
        return next({
            ...args,
            input,
        });
    };
}
exports.idNormalizerMiddleware = idNormalizerMiddleware;
exports.idNormalizerMiddlewareOptions = {
    step: "initialize",
    tags: ["ROUTE53_IDS"],
    name: "idNormalizerMiddleware",
    override: true,
};
const getIdNormalizerPlugin = (unused) => ({
    applyToStack: (clientStack) => {
        clientStack.add(idNormalizerMiddleware(), exports.idNormalizerMiddlewareOptions);
    },
});
exports.getIdNormalizerPlugin = getIdNormalizerPlugin;
